src/unit.rs

```diff
@@ -792,27 +792,31 @@ impl<'input, 'abbrev, 'unit, Endian> DebuggingInformationEntry<'input, 'abbrev,
 
     /// Find the first attribute in this entry which has the given name,
     /// and return it. Returns `Ok(None)` if no attribute is found.
-    pub fn attr(&self, name: constants::DwAt) -> Option<Attribute<'input, Endian>> {
+    pub fn attr(&self, name: constants::DwAt) -> Result<Option<Attribute<'input, Endian>>> {
         let mut attrs = self.attrs();
-        while let Ok(Some(attr)) = attrs.next() {
+        while let Some(attr) = try!(attrs.next()) {
```

Comment:
Agree, but we haven't used ? in gimli yet, so I want to leave that for another PR.